feat: modernize pyproject.toml with comprehensive tool configurations

onlyname0 opened the pull request:
## Summary
- Modernizes pyproject.toml with comprehensive configurations for development tools
- Improves developer experience with better tooling setup
- Ensures consistent code quality standards

## Changes
- Update dev dependencies to use flexible version constraints (>=)
- Add pytest-cov for test coverage, mypy for type checking, and pre-commit for git hooks
- Simplify black configuration with target Python versions
- Add comprehensive pytest configuration with asyncio support
- Add mypy type checking configuration
- Add coverage configuration sections for pytest-cov usage
- Include updated poetry.lock file for reproducible builds

## Test Plan
- [x] Tested with `act` tool locally
- [x] All dependencies install correctly
- [x] Linting passes with existing codebase
- [ ] CI passes on PR

onlyname0 committed:
feat: modernize pyproject.toml with comprehensive tool configurations

- Update dev dependencies to use flexible version constraints (>=)
- Add pytest-cov, mypy, and pre-commit to dev dependencies
- Simplify black configuration with target Python versions
- Add comprehensive pytest configuration with asyncio support
- Add mypy type checking configuration
- Add coverage.py configuration for test coverage reporting

elfjes reviewed (changes requested):
I'm a little bit confused what this PR does? It seems to remove some configuration (black) that may still be needed, and adds some configuration (mypy, coverage) that we're not using. If you want to add coverage or mypy that's ok, but please also update the Makefile and github actions

elfjes commented on pyproject.toml:
```diff
@@ -47,14 +30,46 @@ tqdm = "^4.64.1"
 [tool.pytest.ini_options]
 pythonpath = ["src"]
+addopts = [
+    "-v",
```
interesting how for some PRs you seem to prefer quiet output and for this one you prefer verbose. Please drop the setting entirely since it's just a personal preference

onlyname0 replied:
Across all packages should there be a standard set of `addopts`? `"--tb=short"`, `"--strict-markers"` and `--strict-config`? what about a `minversion` ?

elfjes reviewed (approved):
Approved with small change request

onlyname0 committed:
remove: completely remove mypy from project

- Remove mypy from pyproject.toml dev dependencies
- Remove mypy from Makefile lint target
- Update poetry.lock to reflect dependency changes
- Remove .mypy_cache directory

The linting pipeline now runs with flake8, black, and isort only.
All existing code quality checks continue to pass successfully.

onlyname0 committed:
Update pytest configuration and add maintainers field

- Add pytest minversion requirement (7.0)
- Add --strict-config to pytest addopts
- Add maintainers field to project metadata
